cmds/boot/localboot/main.go

```diff
@@ -0,0 +1,259 @@
+// Copyright 2017-2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"syscall"
+
+	"github.com/u-root/u-root/pkg/bootconfig"
+	"github.com/u-root/u-root/pkg/storage"
+)
+
+// TODO backward compatibility for BIOS mode with partition type 0xee
+// TODO use a proper parser for grub config (see grub.go)
```

Comment:
a proper grub parser is ... a lot. it's just about a programming language. I would rather argue that we should stick with simple grub, since the inventor of the grub configuration language doesn't like it.

Reply:
yeah, we eventually decided to stay with a simple grub parser, it's really too much to handle, especially being a "transition" boot mode